### PR TITLE
Refs #35030 -- Added more tests for @user_passes_test decorator.

### PR DESCRIPTION
**Pull request by felixxm**

### CODE REVIEW

**Comment by felixxm** [comment]
Extra tests that I want to merge before working on #17621.

File: tests/auth_tests/test_decorators.py [hunk 1]
```diff
@@ -142,3 +146,37 @@ def a_view(request):
...
+    @classmethod
+    def setUpTestData(cls):
+        cls.user_pass = models.User.objects.create(username="joe", password="qwerty")
+        cls.user_deny = models.User.objects.create(username="jim", password="qwerty")
+        models.Group.objects.create(name="Joe group")
+        # Add permissions auth.add_customuser and auth.change_customuser
+        perms = models.Permission.objects.filter(
+            codename__in=("add_customuser", "change_customuser")
+        )
+        cls.user_pass.user_permissions.add(*perms)
```
Comment (nessita) [inline tests/auth_tests/test_decorators.py:163]: What's the goal of creating these instances in `setUpTestData` if there is a single test? Do you expect more tests to be added?

**Review by nessita — APPROVED** [review]
LGTM, thanks!

**Comment by felixxm** [comment]
@nessita Thanks for checking :+1:

**Pull request merged**